src/OrbitGl/GraphTrack.h

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_GRAPH_TRACK_H_
+#define ORBIT_GL_GRAPH_TRACK_H_
+
+#include <array>
+#include <map>
+#include <optional>
+#include <string>
+
+#include "Batcher.h"
+#include "CoreMath.h"
+#include "PickingManager.h"
+#include "Timer.h"
+#include "Track.h"
+#include "Viewport.h"
+
+template <size_t Size>
+class TimeSeries {
+ public:
+  explicit TimeSeries(std::array<std::string, Size> names) : names_(names) {}
+
+  [[nodiscard]] const std::array<std::string, Size>& names() const { return names_; }
+  [[nodiscard]] const std::map<uint64_t, std::array<double, Size>>& value_map() const {
+    return value_map_;
+  }
+  [[nodiscard]] bool IsEmpty() const { return value_map_.empty(); }
+  void AddValues(uint64_t timestamp, std::array<double, Size> values) {
+    value_map_[timestamp] = std::move(values);
+  }
```

Comment:
Since std::array is stack-allocated, moving has no performance benefit here. I suggest to use a const-ref as the parameter type and then copy into the map.

Reply:
Changed now. Thanks!